### PR TITLE
Remove shadowing of built-ins

### PR DESCRIPTION
**Pull request by dfawley**

### CODE REVIEW

File: internal/transport/http2_server.go [hunk 1]
```diff
@@ -284,7 +284,7 @@ func newHTTP2Server(conn net.Conn, config *ServerConfig) (_ ServerTransport, err
 }
 
 // operateHeader takes action on the decoded headers.
-func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(*Stream), traceCtx func(context.Context, string) context.Context) (close bool) {
+func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(*Stream), traceCtx func(context.Context, string) context.Context) (connErr bool) {
```
Comment (menghanl) [inline internal/transport/http2_server.go]: Call the returned variable `errored`? Or `failed`?

`connErr` looks like a type `error` to me.

**Comment by dominikh** [comment]
@dfawley Did a default `staticcheck` invocation actually flag these? It shouldn't have unless ST1014 was explicitly enabled; it's a specialised check for security relevant code; shadowing built-ins isn't generally a problem.

**Comment by dfawley** [comment]
@dominikh I think I ran something like `-check '*,-SA1019'` when I got this error (I was just guessing at usage).  I would rather avoid shadowing altogether, so I'm OK with this check.

File: internal/transport/http2_server.go (hunk 1 above)
Comment (dfawley) [reply internal/transport/http2_server.go]: How about `fatal`?

**Pull request merged**